pkg/recommendation/map.go

```diff
@@ -0,0 +1,153 @@
+are arranged by node name.
+
+1. initial stage it will be a list of block devices.
+[]blockdevice
+
+2.
+at last it will be converted into below map -
+- HDD:
+	- node-1 []blockdevice
+	- node-2 []blockdevice
+	- node-3 []blockdevice
+- SSD:
+	- node-1 []blockdevice
+	- node-2 []blockdevice
+	- node-3 []blockdevice
+
+3. In some case all the may not have block device or all the
+types of block device. For that this may look like this -
+- HDD:
+	- node-1 []blockdevice
+	- node-3 []blockdevice
+	- node-4 []blockdevice
+	- node-7 []blockdevice
+- SSD:
+	- node-3 []blockdevice
+	- node-5 []blockdevice
+	- node-7 []blockdevice
+*/
+func buildNodeBlockDeviceMap(
+	bdlist unstructured.UnstructuredList) map[string]map[string][]BlockDeviceInfo {
+	deviceTypeNodeBlockDeviceMap := make(map[string]map[string][]BlockDeviceInfo)
+
+	for _, bd := range bdlist.Items {
+
+		// If block device is not in active state or we got any error while
+		// fetching block device status then we can not use that block device
+		// to create a pool.
+		if ok, err := bdutil.IsActiveOrError(bd); err != nil || !ok {
+			continue
+		}
+
+		// If claim status block device is 'Claimed' or we got any error while
+		// fetching claim status of block device then we can not use that block
+		// device to create a pool.
+		if ok, err := bdutil.IsUnclaimedOrError(bd); err != nil || !ok {
+			continue
+		}
+
+		// If file system present in a block device or we got any error while
+		// fetching file system details of block device then we can not use that
+		// block device to create a pool.
+		if ok, err := bdutil.HasFileSystemOrError(bd); err != nil || ok {
+			continue
+		}
+
+		// Block device should be associated with a node if mode name is missing or
+		// we got any error during fatching node name then we can not use that
+		// block device to create a pool.
+		nodeName, err := bdutil.GetNodeNameOrError(bd)
+		if err != nil {
+			continue
+		}
+
+		// Host name property is a must for Block device. If host name is missing or
+		// we got any error during fatching host name then we can not use that
+		// block device to create a pool.
+		hostName, err := bdutil.GetHostNameOrError(bd)
+		if err != nil {
+			continue
+		}
+
+		// Capacity of a block device is a must to create a recommendation. If capacity
+		// is missing or we got any error during fatching capacity then we can not use
+		// that block device to create a recommendation.
+		capacity, err := bdutil.GetCapacityOrError(bd)
+		if err != nil {
+			continue
+		}
+
+		// Device type of a block device is a must to create a recommendation. If
+		// device type is missing or we got any error during fatching device type
+		// then we can not use it to create a recommendation.
+		deviceType, err := bdutil.GetDeviceTypeOrError(bd)
+		if err != nil {
+			continue
+		}
+
+		// nodeBlockDeviceMap contains block devices grouped by node
+		// If for any device type (HDD OR SSD) this map is not present
```

Comment:
@shovanmaity As discussed, we need a broader type than just the native ones (hdd/ssd). Lets come up with a derived type, using BD properties like native_type-phy_sec_size-log_sec_size. This way, we will be able to support many combinations.